src/ngraph/ngraph_graph.h

```diff
@@ -255,19 +274,48 @@ class Graph : public Node {
   std::vector<int> cached_aux_positions[kGraphExeModeCount];
 
   const bool enable_fprop_cache;
+
+  std::vector<NodePtr> outputs_;
+  std::vector<std::shared_ptr<OutputElement>> output_elements_;
+};
+
+// Element to represent outputs of fused Graphs
```

Comment:
Are there comments anywhere explaining the term "fused graph"?  I'm concerned that it's easy to confuse with "operator fusion", which is also a relevant concept in this overall code.